src/main/java/frc/robot/commands/SwerveJoystickCommand.java

```diff
@@ -0,0 +1,99 @@
+
+    private final Supplier<Double> xSpdFunction, ySpdFunction, turningSpdFunction, speedControlFunction;
+    private final SlewRateLimiter xLimiter, yLimiter, turningLimiter;
+    private final SwerveSubsystem swerveDrivetrain;
+
+    public SwerveJoystickCommand(
+        Supplier<Double> frontBackFunction, Supplier<Double> leftRightFunction, Supplier<Double> turningSpdFunction,
+        Supplier<Double> speedControlFunction, SwerveSubsystem csd) {
+        
+        this.xSpdFunction = frontBackFunction;
+        this.ySpdFunction = leftRightFunction;
+        this.turningSpdFunction = turningSpdFunction;
+        this.speedControlFunction = speedControlFunction;
+        this.xLimiter = new SlewRateLimiter(Constants.kTeleDriveMaxAccelerationUnitsPerSecond);
+        this.yLimiter = new SlewRateLimiter(Constants.kTeleDriveMaxAccelerationUnitsPerSecond);
+        this.turningLimiter = new SlewRateLimiter(Constants.kTeleDriveMaxAngularAccelerationUnitsPerSecond);
+        this.swerveDrivetrain=csd;
+
+        addRequirements(swerveDrivetrain);
+    }
+
+
+    @Override
+    public void initialize() {
+       
+    }
+
+    @Override
+    public void execute() {
+        // 1. Get real-time joystick inputs
+        double xSpeed = xSpdFunction.get(); // xSpeed is actually front back (front +, back -)
+        double ySpeed = ySpdFunction.get(); // ySpeed is actually left right (left +, right -)
+        double turningSpeed = turningSpdFunction.get(); // turning speed is anti-clockwise +, clockwise -
+
+        // 2. Normalize inputs
+        double length = xSpeed * xSpeed + ySpeed * ySpeed; // acutally length squared
+        if (length > 1d) { 
+            length = Math.sqrt(length);
+            xSpeed /= length;
+            ySpeed /= length;
+        }
+
+        // 3. Apply deadband
+        xSpeed = Math.abs(xSpeed) > 0.01 ? xSpeed : 0.0;
+        ySpeed = Math.abs(ySpeed) > 0.01 ? ySpeed : 0.0;
+        turningSpeed = Math.abs(turningSpeed) > 0.01 ? turningSpeed : 0.0;
+
+        // 4. Make the driving smoother
+        double driveSpeed = (Constants.kTeleDriveMaxPercentSpeed - Constants.kTeleDriveMinPercentSpeed)
+        * speedControlFunction.get() + Constants.kTeleDriveMinPercentSpeed;
+
+        xSpeed = xLimiter.calculate(xSpeed) * driveSpeed * Constants.kPhysicalMaxSpeedMetersPerSecond;
+        ySpeed = yLimiter.calculate(ySpeed) * driveSpeed * Constants.kPhysicalMaxSpeedMetersPerSecond;
+        turningSpeed = turningLimiter.calculate(turningSpeed) * driveSpeed * Constants.kPhysicalMaxAngularSpeedRadiansPerSecond;
+        final double x = xSpeed;
+        final double y = ySpeed;
+        final double turn = turningSpeed;
+
+
+        final SwerveRequest.FieldCentric drive = new SwerveRequest.FieldCentric()
+        .withDeadband(Constants.kPhysicalMaxSpeedMetersPerSecond * 0.01).withRotationalDeadband(Constants.kPhysicalMaxAngularSpeedRadiansPerSecond * 0.01) // Add a 10% deadband
```

Comment:
I'm not sure exactly how the .withRotationalDeadband() works but it might be redudant if we already have a deadband from above? Maybe it has more features and is easier to use / debug so we should choose to use this one over the one we applied earlier.